trade-finance-manager-ui/server/controllers/utilisation-reports/check-keying-data/index.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from 'express';
+import { getFormattedReportPeriodWithLongMonth } from '@ukef/dtfs2-common';
+import { FeeRecordPaymentGroup } from '../../../api-response-types';
+import { asUserSession } from '../../../helpers/express-session';
+import api from '../../../api';
+import { CheckKeyingDataViewModel } from '../../../types/view-models';
+import { mapFeeRecordPaymentGroupsToFeeRecordPaymentGroupViewModelItems } from '../helpers';
+
+const renderCheckKeyingDataPage = (res: Response, viewModel: CheckKeyingDataViewModel) => res.render('utilisation-reports/check-keying-data.njk', viewModel);
+
+const getNumberOfFacilities = (feeRecordPaymentGroups: FeeRecordPaymentGroup[]): number =>
+  feeRecordPaymentGroups.reduce((total, { feeRecords }) => feeRecords.length + total, 0);
+
+export const postCheckKeyingData = async (req: Request, res: Response) => {
+  const { userToken, user } = asUserSession(req.session);
+  const { reportId } = req.params;
+
+  try {
+    const { bank, reportPeriod, feeRecordPaymentGroups } = await api.getUtilisationReportReconciliationDetailsById(reportId, userToken);
```

Comment:
See my other comment re grouping but I think we might want to consider a separate endpoint for this fetching specifically the matched fee records in a shape closer to what the front end needs

Reply:
I've added the new endpoint which I think may have doubled the size of this PR 🙃